aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/AwsReadmeDecorator.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package software.amazon.smithy.rustsdk
+
+import org.jsoup.Jsoup
+import software.amazon.smithy.model.traits.DocumentationTrait
+import software.amazon.smithy.rust.codegen.rustlang.raw
+import software.amazon.smithy.rust.codegen.smithy.CodegenContext
+import software.amazon.smithy.rust.codegen.smithy.RustCrate
+import software.amazon.smithy.rust.codegen.smithy.customize.RustCodegenDecorator
+import software.amazon.smithy.rust.codegen.util.getTrait
+
+/**
+ * Generates a README.md for each service crate for display on crates.io.
+ */
+class AwsReadmeDecorator : RustCodegenDecorator {
+    override val name: String = "AwsReadmeDecorator"
+    override val order: Byte = 0
+
+    override fun crateManifestCustomizations(codegenContext: CodegenContext): Map<String, Any?> =
+        mapOf("package" to mapOf("readme" to "README.md"))
+
+    override fun extras(codegenContext: CodegenContext, rustCrate: RustCrate) {
+        rustCrate.withFile("README.md") { writer ->
+            // Strip HTML from the doc trait value. In the future when it's written, we can use our Rustdoc
+            // documentation normalization code to convert this to Markdown.
+            val description = Jsoup.parse(
+                codegenContext.settings.getService(codegenContext.model).getTrait<DocumentationTrait>()?.value ?: ""
+            ).text()
+            val moduleName = codegenContext.settings.moduleName
+
+            writer.raw(
+                """
+                # $moduleName
+
+                **Please Note: The SDK is currently released as an alpha and is intended strictly for
```

Comment:
Long term, I think we should probably generate the entire top-level README.md file in aws-sdk-rust so that it's kept in sync with this one. I omitted a few sections in this README since I didn't think all of them made sense in this context.

For now, we will need to remember to update both READMEs, unless you think the investment is worth it now?

Reply:
maybe we can put an HTML comment in both noting that they both need to be updated

Reply:
I'm skeptical anyone would actually see that when updating one of the READMEs unless they were updating the entire README rather than just a part of it.